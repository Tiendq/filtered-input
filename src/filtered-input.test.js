import React from 'react';
import renderer from 'react-test-renderer';
import numeral from 'numeral';
import FilteredInput from './filtered-input';

describe('FilteredInput', () => {
  test('with only required props', () => {
    const tree = renderer.create(
      <FilteredInput type="text" filterPattern="^[a-z]+$" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('with initial value', () => {
    const tree = renderer.create(
      <FilteredInput type="text" filterPattern="^[a-z]+$" value="Tien" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('with all props', () => {
    const tree = renderer.create(
      <FilteredInput type="text" id="test" name="test" filterPattern="^[a-z]+$" validatePattern="^[a-z]{3}$" onChange={() => true} maxLength="3" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('onChange is always passed if NO validatePattern supplied', () => {
    let onChange = jest.fn();

    const tree = renderer.create(
      <FilteredInput type="text" filterPattern="^[a-z]+$" onChange={onChange} />
    ).toJSON();

    tree.props.onChange({ target: { value: 'tien' }});
    expect(onChange).toHaveBeenCalledWith('tien', true);
  });

  test('onChange called with validatePattern', () => {
    let onChange = jest.fn();

    const tree = renderer.create(
      <FilteredInput type="text" filterPattern="^[a-z]+$" validatePattern="^[a-z]{3}$" onChange={onChange} />
    ).toJSON();

    tree.props.onChange({ target: { value: 'tien' }});
    expect(onChange).toHaveBeenLastCalledWith('tien', false);

    tree.props.onChange({ target: { value: 'hoa' }});
    expect(onChange).toHaveBeenLastCalledWith('hoa', true);
  });

  it('should allow formatting with a customFormatter', () => {
    const customFormatter = (value) => value.length > 0 ? numeral(Number.parseInt(value, 10)).format('0,0') : '';

    const tree = renderer.create(
      <FilteredInput type="text" value="5000" filterPattern="^\d{0,6}$" validatePattern="^\d{1,6}$" customFormatter={customFormatter} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
