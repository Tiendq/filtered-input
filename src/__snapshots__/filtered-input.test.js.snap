exports[`FilteredInput should allow formatting with a customFormatter 1`] = `
<input
  customFormatter={[Function]}
  onChange={[Function]}
  type="text"
  value="5000" />
`;

exports[`FilteredInput with all props 1`] = `
<input
  id="test"
  maxLength="3"
  name="test"
  onChange={[Function]}
  type="text"
  value="" />
`;

exports[`FilteredInput with initial value 1`] = `
<input
  onChange={[Function]}
  type="text"
  value="Tien" />
`;

exports[`FilteredInput with only required props 1`] = `
<input
  onChange={[Function]}
  type="text"
  value="" />
`;
